core/objects_test.go

```diff
@@ -190,7 +190,7 @@ func TestRenewalInfoSuggestedWindowIsWithin(t *testing.T) {
 	test.Assert(t, window.IsWithin(now.Add(time.Minute*30)), "Middle of window should be within the window")
 
 	// Exactly the end, inclusive of the last nanosecond.
```

Comment:
It should state "exclusive of the last nanosecond." or have these two tests:
```
	// End inclusive of the last nanosecond.
	test.Assert(t, !window.IsWithin(now.Add(time.Hour)), "End of window should be outside the window")

	// End exclusive of the last nanosecond.
	test.Assert(t, window.IsWithin(now.Add(time.Hour-time.Nanosecond)), "End of window should be inside the window")
```

You could remove this test now
```
// Just after the last nanosecond.
	test.Assert(t, !window.IsWithin(now.Add(time.Hour+time.Nanosecond)), "After the window should not be within the window")
```